### **Variant Calling and Downstream Analysis of SARS-CoV-2 WGS Datasets**

This project was done as part of Wellcome Connecting Science's "Bioinformatics for Biologists: Analysing and Interpreting Genomics Datasets" course on FutureLearn.

This course employed linux for dataset preprocessing, sequence quality control, genome mapping, and variant calling for identifying variants.
Some of the tools employed were:
1. multiQC
2. Samtools
3. BWA-MEM
4. bcftools
5. SRA-toolkit
6. FreeBayes
7. Integrative Genomics Viewer (IGV)
8. nf-core's viralrecon

Downstream Analysis and Visualization was done using RStudio packages, such as:
1. dplyr
2. tidyverse
3. ggplot2
